### ISSUE TITLE
Docs: consistent-generic-constructors

### ISSUE THREAD

**Issue by burtek** [typescript-eslint/typescript-eslint#5395]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

The [docs](https://typescript-eslint.io/rules/consistent-generic-constructors/) for `constructor` option, state that

```ts
const map: Map<string, number> = new MyMap();
```

is both correct and incorrect.

I think the correct version of it should be changed to below (or just removed as the `Set` example covers this case already)

```ts
const map: Map<string, number> = new MyMap<string, number>();
```

### Affected URL(s)

https://typescript-eslint.io/rules/consistent-generic-constructors/

**Comment by burtek** [comment]
Nevermind, the correct one has `MyMap`. Missed that detail
